### PR TITLE
Add Object Space mapping to NME PerturbNormal block

### LINKED ISSUE #13420: NME PerturbNormal block: add support for object space sampling mode
See https://forum.babylonjs.com/t/read-texture-in-object-space-in-nme/37182

In this mode, we would need to set the `OBJECTSPACE_NORMALMAP` define and pass the `normalMatrix` to the shader code. Which in turns means we have to add a new **world** matrix entry to the `PerturbNormal` block:

https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Materials/PBR/pbrBaseMaterial.ts#L2001-L2004

### PR DESCRIPTION
**Pull request by sebavan**
Fix #13420

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13501/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13501/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13501/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13501/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13501/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13501/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Materials/Node/Blocks/Fragment/perturbNormalBlock.ts [hunk 1]
```diff
@@ -279,6 +303,11 @@ export class PerturbNormalBlock extends NodeMaterialBlock {
         state.compilationString += this._declareOutput(this.output, state) + " = vec4(0.);\r\n";
         state.compilationString += state._emitCodeFromInclude("bumpFragment", comments, {
             replaceStrings: [
+                { search: /texture2D\(bumpSampler,vBumpUV\)/g, replace: `${uvForPerturbNormal}` },
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Node/Blocks/Fragment/perturbNormalBlock.ts:306]: Was that a bug in the existing code?

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Materials/Node/Blocks/Fragment/perturbNormalBlock.ts (hunk 1 above)
Comment (sebavan) [reply packages/dev/core/src/Materials/Node/Blocks/Fragment/perturbNormalBlock.ts:306]: nope, this block is only in the case of object space :-) the other one has a + uvOffset

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13501/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Pull request merged**